File: src/1603.design-parking-system.1708/solution.ts
class ParkingSystem {
    constructor (
        public big: number,
        public medium: number,
        public small: number
    ) {
    }

    addCar (carType: number): boolean {
        if (carType === 1) {
            if (this.big === 0) {
                return false;
            }
            this.big--;
        } else if (carType === 2) {
            if (this.medium === 0) {
                return false;
            }
            this.medium--;
        } else {
            if (this.small === 0) {
                return false;
            }
            this.small--;
        }
        return true;
    }
}

/**
 * Your ParkingSystem object will be instantiated and called as such:
 * var obj = new ParkingSystem(big, medium, small)
 * var param_1 = obj.addCar(carType)
 */

File: src/1603.design-parking-system.1708/solution.java
class ParkingSystem {
    private int big;
    private int medium;
    private int small;

    public ParkingSystem(int big, int medium, int small) {
        this.big = big;
        this.medium = medium;
        this.small = small;

    }

    public boolean addCar(int carType) {
        if (carType == 1) {
            if (big > 0) {
                this.big--;
                return true;
            }
            return false;
        }
        if (carType == 2) {
            if (this.medium > 0) {
                this.medium--;
                return true;
            }
            return false;
        }
        if (this.small > 0) {
            this.small--;
            return true;
        }
        return false;
    }
}

/**
 * Your ParkingSystem object will be instantiated and called as such:
 * ParkingSystem obj = new ParkingSystem(big, medium, small);
 * boolean param_1 = obj.addCar(carType);
 */